src/dataEditor/dataEditorClient.ts

```diff
@@ -94,8 +96,19 @@ const OMEGA_EDIT_MIN_PORT: number = 1024
 // file-scoped types
 // *****************************************************************************
 
+class ServerInfo implements IServerInfo {
+  serverHostname: string = 'unknown' // hostname
```

Comment:
I've been reading a bit on *stringly-typed* objects in TypeScript and it seems to be best to avoid them
> See **Item 33: Prefer More Precise Alternatives to String Types** in *Effective TypeScript* by Don Vanderkam

Maybe `serverHostname: string` could be `serverHostname: Hostname` where <code>type Hostname = "unknown" | "localhost" | \`{number}.{number}.{number}.{number}.` </code>. Likewise for other

Reply:
Seems like that would add a lot of boilerplate to the code for very little benefit.

Reply:
I'd have to look more at the flow of data for this particular instance but just in general I like the idea of strict and defined types over `string` or `number`. There are also cases where you can "type define" yourself into a corner and this might be one of them.